src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModel.h

```diff
@@ -0,0 +1,256 @@
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_ESTIMATORS_SUBMODEL_H
+#define BIPEDAL_LOCOMOTION_ESTIMATORS_SUBMODEL_H
+
+#include <vector>
+#include <string>
+#include <memory>
+
+// iDynTree
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+#include <iDynTree/KinDynComputations.h>
+
+// BLF
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+namespace RobotDynamicsEstimator
+{
+
+/**
+ * Sensor describes the generic sensors of the submodels.
+ */
+struct Sensor
+{
+    std::string name; /**< Name of the sensor in the URDF */
+    std::string frame; /**< Frame of the sensor in the URDF */
+};
+
+/**
+ * FT describes the force/torque sensors of the submodels.
+ */
+struct FT : Sensor
+{
+    enum class Direction : int { positive=1, negative=-1, notspecified=0 };
+    Direction forceDirection = Direction::notspecified; /**< Force direction depending on which side of the sensor is considered (+1 or -1)*/
+};
+
+/**
+ * SubModel is a concrete class describing the sub-model object, its model, the list of sensors and the mapping between
+ * its joints and the same joints in the full model.
+ */
+class SubModel
+{
+private:
+    iDynTree::Model model; /**< iDynTree Model object describing the submodel */
+    std::vector<int> jointListMapping; /**< Each element contains an index describing the joint position in the full model*/
+    std::vector<FT> ftList; /**< List of force/torque sensors in the submodel */
+    std::vector<Sensor> accelerometerList; /**< List of accelerometers in the submodel */
+    std::vector<Sensor> gyroscopeList; /**< List of gyroscopes in the submodel */
+    std::vector<std::string> externalContactList; /**< List of the additional external contacts */
+
+public:
+    /**
+     * Getters
+     */
+
+    /**
+    * @brief Access model.
+    * @return The model of the SubModel.
+    */
+    iDynTree::Model getModel();
+
+    /**
+     * @brief Access jointListMapping.
+     * @return the mapping between the joint indeces in the sub-model and the joint indeces in the full-model.
+     */
+    std::vector<int> getJointMapping();
+
+    /**
+     * @brief Access ftList.
+     * @return the list of FT objects which is the list of force/torque sensors.
+     */
+    std::vector<FT> getFTList();
+
+    /**
+     * @brief Access accelerometerList.
+     * @return a list of Sensor objects describing the accelerometers contained in the sub-model.
+     */
+    std::vector<Sensor> getAccelerometerList();
+
+    /**
+     * @brief Access gyroscopeList.
+     * @return a list of Sensor objects describing the gyroscope contained in the sub-model.
+     */
+    std::vector<Sensor> getGyroscopeList();
+
+    /**
+     * @brief Access externalContactList.
+     * @return a list of strings describing frame names of the external contacts for the sub-model.
+     */
+    std::vector<std::string> getExternalContactList();
```

Comment:
Here you can output all these by const reference and make the methods const, e.g.
```cpp
const iDynTree::Model& getModel() const;
```

The output const reference avoids creating a temporary object every time the method is called. In fact, right now, it creates a brand new model/vector at every call. This can be problematic if the output requires a lot of memory.

The ``const`` after the method name is to notify that the method can be called even when you have a const reference to the class, because you are making explicit the fact that by calling the method, the internal state does not change.